src/open_inwoner/utils/decorators.py

```diff
@@ -50,19 +81,39 @@ def wrapped(*args, **kwargs) -> RT:
                 }
                 key_kwargs[argspec.varkw] = var_kwargs
 
-            cache_key = key.format(**key_kwargs)
-
-            _cache = caches[alias]
+            (
+                cache_key_with_attr_placeholders,
+                attr_mapping,
+            ) = _map_cache_key_instance_attrs_to_placeholders(key)
+            if attr_mapping:
+                if len(args) == 0:
+                    raise ValueError(
+                        "You can only access attributes on `self` when decorating a method"
+                    )
+
+                bound_instance = args[0]
+                for mapped_attr, original_attr in attr_mapping.items():
+                    try:
+                        key_kwargs[mapped_attr] = getattr(bound_instance, original_attr)
+                    except AttributeError:
+                        raise AttributeError(
+                            f"Attribute `{original_attr}` does not exist on bound instance"
+                        )
```

Comment:
You can also add a note to Python's `AttributeError` instead of raising a new one:
```
except AttributeError as exc:
    exc.add_note(f"Attribute `{original_attr}` does not exist on bound instance")
    raise
```